### PR TITLE
Allow to add the showcase view to a custom parent

### PR DESCRIPTION
**Pull request by martinbonnin**
I needed this because at the startup of my app, the LoginFragment competes with the ShowcaseView to be on top of everything else in the Activity

### CODE REVIEW

File: library/src/main/java/com/github/amlcurran/showcaseview/ShowcaseView.java [hunk 1]
```diff
@@ -387,6 +387,9 @@ public void showButton() {
         final ShowcaseView showcaseView;
         private final Activity activity;
 
+        private ViewGroup mParent;
```
Comment (amlcurran) [inline library/src/main/java/com/github/amlcurran/showcaseview/ShowcaseView.java]: Could you remove the `m` prefix from these fields please? :)

File: library/src/main/java/com/github/amlcurran/showcaseview/ShowcaseView.java [hunk 2]
```diff
@@ -403,7 +406,16 @@ public Builder(Activity activity, boolean useNewStyle) {
          * @return the created ShowcaseView
          */
         public ShowcaseView build() {
-            insertShowcaseView(showcaseView, activity);
+            ViewGroup parent;
+            int index;
+            if (mParent == null) {
+                parent = ((ViewGroup) activity.getWindow().getDecorView());
```
Comment (amlcurran) [inline library/src/main/java/com/github/amlcurran/showcaseview/ShowcaseView.java]: you could assign this to `mParent` in the constructor of Builder, and then you wouldn't have you check against null?

**Comment by amlcurran** [comment]
Like the idea of this, just a couple of issues, and a conflict, but happy to merge once these have been addressed :)

**Comment by martinbonnin** [comment]
thanks for the comments. Should be better now.

**Comment by amlcurran** [comment]
Looks great! There are still conflicts; just make sure you merge the master branch into your branch, and then that should work :+1:

**Comment by martinbonnin** [comment]
Should be good now, tell me if anything else is required.

**Comment by amlcurran** [comment]
Awesome, thank you very much! I will make a release containing shortly :)

**Pull request merged**